src/Swr/swrRender.c

```diff
@@ -0,0 +1,207 @@
+#include "swrRender.h"
+
+#include <macros.h>
+
+// 0x00409290
+void rdModel3Mesh_CalculateSurfaceNormals(rdModel3Mesh* mesh)
+{
+    HANG("TODO");
+}
+
+// 0x00409380
+void rdFace_ConfigureFogStartEnd(int16_t start, int16_t end)
+{
+    HANG("TODO");
+}
+
+// 0x00409450
+void Direct3d_ConfigFog2(short r, short g, short b, short a)
+{
+    HANG("TODO");
+}
+
+// 0x004094E0
+int rdFace_SetFogEnabled(int)
+{
+    HANG("TODO");
+}
+
+// 0x00409510
+void rdCamera_SetLights(int light_index, int num_lights)
+{
+    HANG("TODO");
+}
+
+// 0x00409600
+float* SetLightColorsAndDirection(int light_index, short ambient_r, short ambient_g, short ambient_b, short light_r, short light_g, short light_b, short pos_x, short pos_y, short pos_z)
+{
+    HANG("TODO");
+}
+
+// 0x00409700
+int SetLightColorsAndDirectionFromPrimaryLight(short light_index)
+{
+    HANG("TODO");
+}
+
+// 0x00409750
+float* SetAlternativeLightColorsAndDirection(int light_index, short light_r, short light_g, short light_b, short pos_x, short pos_y, short pos_z)
+{
+    HANG("TODO");
+}
+
+// 0x00432B80
+void rdModel3Mesh_ApplySwrModelColors(rdModel3Mesh* rdmesh, swrModel_Mesh* mesh)
+{
+    HANG("TODO");
+}
+
+// 0x00432D30
+void rdModel_ConvertSwrModelMesh(Gfx* display_list, rdModel3Mesh* result, swrModel_Mesh* mesh, RdFaceFlag material_flags)
+{
+    HANG("TODO");
+}
+
+// 0x0044C440
+char rdModel_ApplyNodeSettings(char, short)
+{
+    HANG("TODO");
+}
+
+// 0x0044C4C0
+void rdModel_RevertNodeSettings(char)
+{
+    HANG("TODO");
+}
+
+// 0x0044c510
+int16_t AABBFrustumTest(float* aabb, bool full_test)
+{
+    HANG("TODO");
+}
+
+// 0x0044ca00
+void rdModel_Add3064NodeToScene(swrModel_Node* node)
+{
+    HANG("TODO");
+}
+
+// 0x0044ceb0
+void rdModel_Add3064NodeToScene2(swrModel_Node* node)
+{
+    HANG("TODO");
+}
+
+// 0x0044d1e0
+void ApplyD065PivotTransform(rdMatrix34* inout_transform, const rdMatrix34* transform, const rdVector3* pivot)
+{
+    HANG("TODO");
+}
```

Comment:
I was wondering if you could rename the `3064` and `D065` flags in the name of these functions and the  one below since it looks like you found what the flags were doing.

Reply:
fixed 👍

Reply:
Looks good to me !